packages/core/test/typescript_definitions/js-joda-tests.ts

```diff
@@ -732,3 +727,40 @@ function test_Temporal() {
   temporal.with(nextYear);
   temporal.with(ChronoField.YEAR, 2020);
 }
+
+function test_TemporalQuery() {
+    const temporal1: Temporal = Instant.now();
+    const temporal2: Instant = Instant.now();
+    const temporal3: LocalDateTime = LocalDateTime.now();
+    const temporal4: ZonedDateTime = ZonedDateTime.now();
+
+    const chronology = temporal1.query(TemporalQueries.chronology());
+
+    const unit = temporal2.query(TemporalQueries.precision());
+    if (unit) {
+        temporal1.plus(1, unit);
+    }
+
+    const localDate = temporal3.query(TemporalQueries.localDate());
+    if (localDate) {
+        localDate.atStartOfDay();
+    }
+
+    const zoneId = temporal4.query(TemporalQueries.zoneId());
+    if (zoneId) {
+        temporal3.atZone(zoneId);
+    }
+
```

Comment:
That looks strange, looks like there is null returned and the tests in the if statement are not executed.

I played around a bit, whats about something like this ?
```
    const instant: Instant = Instant.now();
    const localDateTime: LocalDateTime = LocalDateTime.now();
    const zonedDateTime: ZonedDateTime = ZonedDateTime.now();


    const unit = instant.query(TemporalQueries.precision());
    expectType<TemporalUnit>(unit);

    const localDate = localDateTime.query(TemporalQueries.localDate());
    expectType<LocalDate>(localDate);

    const zoneId = zonedDateTime.query(TemporalQueries.zoneId());
    expectType<ZoneId>(zoneId);
```

Reply:
Well, as you mentioned before the file is never executed, instead it is just checked by TS so every path is analyzed by the compiler.

Right now you don't have the `strictNullChecks` flag enabled, if it would then TS will notice that the type is possible `null` but it'll narrow the type automatically inside the `if` block at compile time since the control flow analysis "knows" the only way to enter inside that block is when `unit` is not null. I made the test like that since was more natural to me, is more similar to the way I think users will use it and it will not break in case you should decide to enable `strictNullChecks` in the future. The compiler still checks the path inside the `if`, the only difference is that it knows there the type cannot be `null`.

If you still prefer to me to change it I have no problem!

Reply:
To be honest, i would prefer to change it. Like you did its quite hard to read in my opinion and it doesn't fit to the patterns we use in the rest of the typscript test.

Reply:
Sure, you're the boss :wink:

Reply:
At the end you do have `strictNullChecks` enabled so I have to assert the exact type outside the `if` block. But change is done.